### ISSUE TITLE
irb prompt in git-bash disappeared (Windows)

### ISSUE THREAD

**Issue by cazyw** [git-for-windows/git#655]
When I upgraded from an old version of git-bash to the latest version the irb prompts all disappeared.

In git-bash all I see is 
`Switch to inspect mode.`
`followed by a blinking cursor`

However when I use the Windows command prompt I still see
` irb(main):001:0>`

Wasn't sure if this was a known issue or if there's something wrong with my setup.

**Comment by shiftkey** [comment]
Please provide more details about your setup - in particular:

 - what version of Git for Windows was working for you
 - what version of Git for Windows has this issue
 - what version of Ruby you have installed - I gather this is obtainable from http://rubyinstaller.org/

**Comment by dscho** [comment]
> I upgraded from an old version

Could that be more specific, please?

**Comment by dscho** [comment]
Also, from [our release notes](https://github.com/git-for-windows/build-extra/blob/master/installer/ReleaseNotes.md#known-issues):

> Some console programs interact correctly with MinTTY only when called through `winpty` (e.g. the Python console needs to be started as `winpty python` instead of just `python`).

Does this apply to your case, too?

**Comment by cazyw** [comment]
Thanks all, I tried a few things and fixed the problem. Sorry my info was rather sparse....

Thanks dscho it was MinTTY that was causing issues.

I'm on Windows 7 and am currently on
ruby 2.2.4p230 (2015-12-16 revision 53155) [i386-mingw32]
Rails 4.2.5.1

Using git for windows version 1.9.5.msysgit.1 (File: Git-1.9.5-preview20150319.exe) was ok.

When I upgraded to git version 2.7.1.windows.2 (File: Git-2.7.1.2-64-bit.exe) there were two options during the installation. Selecting the option (during installation):

- "Use Windows' default console window" --> everything okay, `irb(main):001:0>` appears
- "Use MinTTY (the default terminal of MSys2)" --> only blinking cursor appears.

Creating and adding the line `IRB.conf[:PROMPT_MODE] = :DEFAULT` to ~/.irbrc added the prompt back in and now running `irb` and `rails console` displays the prompt.
